lib/mix_test_watch/run.ex

```diff
@@ -0,0 +1,43 @@
+defmodule MixTestWatch.Run do
+  @moduledoc """
+  Runs tests according to given path
+  """
+
+  alias MixTestWatch.Command
+  alias MixTestWatch.Files
+  alias MixTestWatch.Shell
+
+  @spec run(String.t, String.t) :: :ok
+
+  def run(path, config) do
+    path
+    |> Files.find_test
+    |> case do
+      :none       -> run_all(config)
+      {:ok, one } -> run_one(one, config)
+    end
+    |> case do
+      {:one, :ok} -> run_all(config)
+      {_, res}    -> res
+    end
+  end
+
+  @spec run_tests(String.t) :: :ok
+
+  def run_tests(config) do
+    config |> Command.build |> Shell.exec
+  end
+
+  @spec run_one(String.t, String.t) :: :ok
+
```

Comment:
Are these the correct typing for the spec? Maybe I'm mis-reading (or a little naive when it comes to typespecs) but it looks like we're returning a tuple below with `{:one, ... }` but in the typespec above defines the return as `:ok`. Is that correct?